src/api/entities/CorporateBallot/index.ts

```diff
@@ -0,0 +1,298 @@
+      assetId: asset.id,
+    });
+  }
+
+  /**
+   * Return the status of the Ballot
+   *
+   * @throws if the Ballot does not exist
+   */
+  public async status(): Promise<CorporateBallotStatus> {
+    const details = await this.details();
+
+    return ballotDetailsToBallotStatus(details);
+  }
+
+  /**
+   * Retrieve the results of the Ballot
+   *
+   * @throws if the Ballot does not exist
+   */
+  public async results(): Promise<CorporateBallotMetaWithResults> {
+    const {
+      id,
+      asset,
+      context,
+      context: {
+        polymeshApi: { query },
+      },
+    } = this;
+
+    const {
+      meta: { title: ballotTitle, motions },
+    } = await this.details();
+
+    const caId = corporateActionIdentifierToCaId({ localId: id, asset }, context);
+
+    const rawResults = await query.corporateBallot.results(caId);
+    const results = rawResults.map(result => u128ToBigNumber(result));
+
+    const metaWithResults: CorporateBallotMetaWithResults = {
+      title: ballotTitle,
+      motions: [],
+    };
+
+    let resultIndex = 0;
+
+    motions.forEach(({ title, infoLink, choices }) => {
+      const motionWithResults: CorporateBallotMotionWithResults = {
+        title,
+        infoLink,
+        choices: [],
+        total: new BigNumber(0),
+      };
+
+      choices.forEach(choice => {
+        motionWithResults.choices.push({
+          choice,
+          votes: results[resultIndex],
+        });
+        motionWithResults.total = motionWithResults.total.plus(results[resultIndex]);
+        resultIndex += 1;
+      });
+
+      metaWithResults.motions.push(motionWithResults);
+    });
+
+    return metaWithResults;
+  }
+
+  /**
+   * Retrieve the participation of the Ballot
+   *
+   * @throws if the Ballot does not exist
+   */
+  public async votesByIdentity(did: Identity | string): Promise<CorporateBallotWithParticipation> {
+    const {
+      id,
+      asset,
+      context,
+      context: {
+        polymeshApi: { query },
+      },
+    } = this;
+
+    const {
+      meta: { title: ballotTitle, motions },
+    } = await this.details();
+
+    const caId = corporateActionIdentifierToCaId({ localId: id, asset }, context);
+    const identityId = stringToIdentityId(asIdentity(did, context).did, context);
+
+    const rawDidVotes = await query.corporateBallot.votes(caId, identityId);
+
+    const voteMap = new Map<number, BigNumber>();
+    const fallbackMap = new Map<number, string | undefined>();
+    const choicesMap = new Map<number, string>();
+
+    let choiceIndex = 0;
+
+    motions.forEach(({ choices }) => {
+      choices.forEach(choice => {
+        choicesMap.set(choiceIndex, choice);
+        choiceIndex += 1;
+      });
+    });
+
+    rawDidVotes.forEach(({ power, fallback }, index) => {
+      voteMap.set(index, u128ToBigNumber(power));
+
+      fallbackMap.set(
+        index,
+        fallback.isSome ? choicesMap.get(u16ToBigNumber(fallback.unwrap()).toNumber()) : undefined
+      );
```

Comment:
The value of fallback corresponds to the index of choice within a singe motion. The current logic will take the value from overall choice list across all motions